curvefs/src/client/s3/disk_cache_manager.cpp

```diff
@@ -331,23 +335,23 @@ void DiskCacheManager::TrimCache() {
             std::string cacheWriteFullDir;
             cacheReadFullDir = GetCacheReadFullDir();
             cacheWriteFullDir = GetCacheWriteFullDir();
-            std::set<std::string> cachedObjNameTmp;
-            {
-                std::lock_guard<bthread::Mutex> lk(mtx_);
-                cachedObjNameTmp = cachedObjName_;
-            }
+
             while (!IsDiskCacheSafe()) {
                 std::string cacheReadFile, cacheWriteFile;
-                std::set<std::string>::iterator cacheKeyIter;
-                cacheKeyIter = cachedObjNameTmp.begin();
-                if (cacheKeyIter == cachedObjNameTmp.end()) {
+                if (cachedObjName_->Size() == 0) {
                     VLOG(3) << "remove disk file error"
-                               << ", cachedObjName is empty.";
+                            << ", cachedObjName is empty.";
                     break;
                 }
-
-                cacheReadFile = cacheReadFullDir + "/" + *cacheKeyIter;
-                cacheWriteFile = cacheWriteFullDir + "/" + *cacheKeyIter;
+                std::string cacheKey;
+                cachedObjName_->GetBack(&cacheKey);
+                if (cacheKey.empty()) {
+                    VLOG(3) << "cachekey is empty";
+                    break;
+                }
+                VLOG(9) << "obj will be removed: " << cacheKey;
+                cacheReadFile = cacheReadFullDir + "/" + cacheKey;
+                cacheWriteFile = cacheWriteFullDir + "/" + cacheKey;
```

Comment:
If cacheWriteFile exists, why not evict the next？

Reply:
may fix later, the issue is : https://github.com/opencurve/curve/issues/1117